PlanPluginBridge/src/main/java/com/djrapitops/pluginbridge/plan/nucleus/NucleusData.java

```diff
@@ -0,0 +1,311 @@
+import com.djrapitops.plan.utilities.html.HtmlUtils;
+import com.djrapitops.plan.utilities.html.icon.Color;
+import com.djrapitops.plan.utilities.html.icon.Family;
+import com.djrapitops.plan.utilities.html.icon.Icon;
+import io.github.nucleuspowered.nucleus.api.NucleusAPI;
+import io.github.nucleuspowered.nucleus.api.nucleusdata.*;
+import io.github.nucleuspowered.nucleus.api.service.*;
+import java.util.*;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.entity.living.player.User;
+import org.spongepowered.api.service.user.UserStorageService;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.text.serializer.TextSerializers;
+
+/**
+ * PluginData for Nucleus plugin.
+ *
+ * @author Vankka
+ */
+public class NucleusData extends PluginData {
+    private UserStorageService userStorageService = null;
+
+    public NucleusData() {
+        super(ContainerSize.TWO_THIRDS, "Nucleus");
+        setPluginIcon(Icon.called("flask").of(Color.DEEP_ORANGE).build());
+
+        Sponge.getServiceManager().provide(UserStorageService.class).ifPresent(storageService -> userStorageService = storageService);
+    }
+
+    @Override
+    public InspectContainer getPlayerData(UUID uuid, InspectContainer inspectContainer) {
+        User user = getUser(uuid);
+
+        if (user == null) {
+            inspectContainer.addValue("Data unavailable", "Could not get user data");
+            return inspectContainer;
+        }
+
+        if (NucleusAPI.getMuteService().isPresent()) {
+            NucleusMuteService muteService = NucleusAPI.getMuteService().get();
+
+            boolean muted = muteService.isMuted(user);
+            inspectContainer.addValue(getWithIcon("Muted", Icon.called("bell-slash").of(Color.DEEP_ORANGE)), muted ? "Yes" : "No");
+
+            if (muted && muteService.getPlayerMuteInfo(user).isPresent()) {
+                MuteInfo muteInfo = muteService.getPlayerMuteInfo(user).get();
+
+                String reason = HtmlUtils.swapColorsToSpan(muteInfo.getReason());
+                String start = "Unknown";
+                String end;
+                String link = "Unknown";
+
+                if (muteInfo.getMuter().isPresent()) {
+                    User operatorUser = getUser(muteInfo.getMuter().get());
+                    if (operatorUser != null) {
+                        String operator = operatorUser.getName();
+                        link = Html.LINK.parse(PlanAPI.getInstance().getPlayerInspectPageLink(operator), operator);
+                    }
+                }
+
+                if (muteInfo.getCreationInstant().isPresent()) {
+                    start = FormatUtils.formatTimeStampYear(muteInfo.getCreationInstant().get().toEpochMilli());
+                }
+
+                if (muteInfo.getRemainingTime().isPresent()) {
+                    end = FormatUtils.formatTimeStampYear(muteInfo.getRemainingTime().get().plusMillis(System.currentTimeMillis()).toMillis());
```

Comment:
Can be simplified with Optional#map
`String end = muteInfo.getRemainingTime().map(instant -> FormatUtils.formatTimeStampYear(instant.toEpochMilli())).orElse("Permanent mute")`

Additionally since this seems to be done often it could be split into it's own method (orElse String as parameter)